packages/feedback/src/screenshot/components/ScreenshotEditorv2.tsx

```diff
@@ -0,0 +1,356 @@
+  const imgClientWidth = measurementDiv.clientWidth;
+  const ratio = imageSource.width / imageSource.height;
+  let width = imgClientHeight * ratio;
+  let height = imgClientHeight;
+  if (width > imgClientWidth) {
+    width = imgClientWidth;
+    height = imgClientWidth / ratio;
+  }
+  const x = (imgClientWidth - width) / 2;
+  const y = (imgClientHeight - height) / 2;
+  return { action: '', x: x, y: y, width: width, height: height };
+};
+
+function drawRect(rect: Rect, ctx: CanvasRenderingContext2D, scale: number = 1, lineWidth: number = 4): void {
+  const scaledX = rect.x * scale;
+  const scaledY = rect.y * scale;
+  const scaledWidth = rect.width * scale;
+  const scaledHeight = rect.height * scale;
+
+  // creates a shadow around
+  ctx.shadowColor = 'rgba(0, 0, 0, 0.7)';
+  ctx.shadowBlur = 50; // Amount of blur for the shadow
+
+  switch (rect.action) {
+    case 'highlight':
+      // draws a rectangle first so that the shadow is visible before clearing
+      ctx.fillStyle = 'rgb(0, 0, 0)';
+      ctx.fillRect(scaledX, scaledY, scaledWidth, scaledHeight);
+
+      ctx.clearRect(scaledX, scaledY, scaledWidth, scaledHeight);
+
+      break;
+    case 'hide':
+      ctx.fillStyle = 'rgb(0, 0, 0)';
+      ctx.fillRect(scaledX, scaledY, scaledWidth, scaledHeight);
+
+      break;
+    default:
+      break;
+  }
+
+  // Disable shadow after the action is drawn
+  ctx.shadowColor = 'transparent';
+  ctx.shadowBlur = 0;
+
+  ctx.strokeStyle = '#ff0000';
+  ctx.lineWidth = lineWidth;
+  ctx.strokeRect(scaledX + 1, scaledY + 1, scaledWidth - 2, scaledHeight - 2);
+}
+
+function resizeCanvas(canvas: HTMLCanvasElement, imageDimensions: Rect): void {
+  canvas.width = imageDimensions.width * DPI;
+  canvas.height = imageDimensions.height * DPI;
+  canvas.style.width = `${imageDimensions.width}px`;
+  canvas.style.height = `${imageDimensions.height}px`;
+}
+
+export function ScreenshotEditorFactoryv2({
+  h,
+  hooks,
+  imageBuffer,
+  dialog,
+  options,
+}: FactoryParams): ComponentType<Props> {
+  const useTakeScreenshot = useTakeScreenshotFactory({ hooks });
+  const Toolbarv2 = ToolbarFactoryv2({ h });
+  const IconClose = IconCloseFactory({ h });
+  return function ScreenshotEditor({ onError }: Props): VNode {
+    const styles = hooks.useMemo(() => ({ __html: createScreenshotInputStyles(options.styleNonce).innerText }), []);
+
+    const [action, setAction] = hooks.useState<'highlight' | 'hide' | ''>('');
+    const [drawCommands, setDrawCommands] = hooks.useState<Rect[]>([]);
+    const [currentRect, setCurrentRect] = hooks.useState<Rect | undefined>(undefined);
+    const measurementRef = hooks.useRef<HTMLDivElement>(null);
+    const screenshotRef = hooks.useRef<HTMLCanvasElement>(null);
+    const graywashRef = hooks.useRef<HTMLCanvasElement>(null);
+    const rectDivRef = hooks.useRef<HTMLDivElement>(null);
+    const [imageSource, setimageSource] = hooks.useState<HTMLCanvasElement | null>(null);
```

Comment:
fix typo, and we can set a reasonable default to avoid so many null checks 

```suggestion
    const [imageSource, setImageSource] = hooks.useState<HTMLCanvasElement>(() => DOCUMENT.createElement('canvas'));
```